### PR TITLE
feat(CreateProfilePage): Build create profile page - FA-37

### PR DESCRIPTION
**Pull request by lgmarchi**
## Description

1. Built the create profile page and its components 
   - Add Image Picker to user take a image from camera or camera roll
   - Add Permission Handler to handle with the device's permissions 
   - Add Image Cropper, for users from Android and IOS crop their image to Uplink
https://user-images.githubusercontent.com/63157656/170344883-0d6fac86-725a-4375-a7dd-c5437038c2b9.mov

3. Refactor the screenshot for recovery seed words
Screen Shot 2022-05-25 at 15 41 01

4. Add a saved dialog for give to user a feedback about the screenshot
Screen Shot 2022-05-25 at 15 42 01

5. Add opacity animation on UButtons when they are disabled. 
image

6. Add onBackPressed function to UAppBar. 

## Type of Change

- [ x ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ x ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: UAppBar.back(title: 'Recovery Seed'),
+      body: SafeArea(
+        child: Padding(
+          padding: const EdgeInsets.fromLTRB(16, 0, 16, 0),
+          child: ListView(
+            children: [
+              const UText(
+                'Write this down in the order that they appear. Having the correct order is crucial when you are recovering your account.',
```
Comment (stavares843) [inline lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart:32]: can we add this to a strings file instead?

File: lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                textStyle: UTextStyle.B1_body,
+              ),
+              const SizedBox(height: 24),
+              const _RecoverySeedBoxes(),
+              const SizedBox(height: 48),
+              UButton.primary(
+                label: 'I Saved It',
```
Comment (stavares843) [inline lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart:39]: can we add this to a strings file instead?

File: lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                          ..pop()
+                          ..push(
+                            MaterialPageRoute<void>(
+                              builder: (context) =>
+                                  const OnboardPrivacySettingFirstPage(),
+                            ),
+                          );
+                      },
+                      header: '''
+              Are you sure you want to continue
```
Comment (stavares843) [inline lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart:59]: can we add this to a strings file instead?

File: lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                    context,
+                    widget: const _ScreenShotRecoverySeedBoxes(),
+                    delay: const Duration(milliseconds: 200),
+                  );
+
+                  if (!mounted) return;
+                  await UBottomSheetTwoButtons(
+                    context,
+                    header: '''
+Screenshot captured!
```
Comment (stavares843) [inline lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart:90]: can we add this to a strings file instead?

File: lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+          ),
+        ),
+      ),
+    );
+  }
+
+  UDialog _savedScreenShotDialog(BuildContext context) {
+    return UDialog(
+      context,
+      title: 'Saved!',
```
Comment (stavares843) [inline lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart:124]: can we add this to a strings file instead?

File: lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+      actions: [],
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          const UText(
+            'This screenshot was saved to your phone gallery.',
```
Comment (stavares843) [inline lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart:130]: can we add this to a strings file instead?

File: lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+            textStyle: UTextStyle.B1_body,
+          ),
+          const SizedBox.square(
+            dimension: 16,
+          ),
+          Row(
+            children: [
+              Expanded(
+                child: UButton.filled1(
+                  label: 'Okay',
```
Comment (stavares843) [inline lib/auth/onboard_recovery_seed_page/onboard_recovery_seed_page.dart:140]: can we add this to a strings file instead?

File: lib/auth/onboarding_create_profile_page.dart [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+                    hintText: 'Digit your status message...',
+                    onChanged: (value) {},
+                    controller: _messageStatusTextFieldController,
+                  ),
+                  const SizedBox.square(dimension: 56),
+                  ValueListenableBuilder(
+                    valueListenable: _isSignInButtonEnabled,
+                    builder: (context, object, widget) {
+                      return UButton.primary(
+                        label: 'Sign in',
```
Comment (stavares843) [inline lib/auth/onboarding_create_profile_page.dart:99]: can we add this to a strings file instead?

File: lib/auth/onboarding_create_profile_page.dart [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+                      value.isNotEmpty
+                          ? _isSignInButtonEnabled.value = true
+                          : _isSignInButtonEnabled.value = false;
+                    },
+                    controller: _usernameTextFieldController,
+                  ),
+                  const SizedBox.square(dimension: 24),
+                  _TextField(
+                    textFieldTitle: 'Status Message',
+                    hintText: 'Digit your status message...',
```
Comment (stavares843) [inline lib/auth/onboarding_create_profile_page.dart:90]: can we add this to a strings file instead?

File: lib/auth/onboarding_create_profile_page.dart [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+                  Align(
+                    child: UUserPictureChange(
+                      onPictureSelected: (pictureSelected) {
+                        _userPicture = pictureSelected;
+                      },
+                    ),
+                  ),
+                  const SizedBox(height: 56),
+                  _TextField(
+                    textFieldTitle: 'Username',
```
Comment (stavares843) [inline lib/auth/onboarding_create_profile_page.dart:78]: can we add this to a strings file instead?

File: lib/auth/onboarding_create_profile_page.dart [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+                    hintText: 'Digit your username...',
```
Comment (stavares843) [inline lib/auth/onboarding_create_profile_page.dart:79]: can we add this to a strings file instead?

File: lib/auth/onboarding_create_profile_page.dart [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+            padding: const EdgeInsets.fromLTRB(16, 0, 16, 0),
+            child: SingleChildScrollView(
+              controller: _scrollController,
+              reverse: true,
+              physics: const BouncingScrollPhysics(),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: [
+                  const SizedBox(height: 16),
+                  const UText(
```
Comment (stavares843) [inline lib/auth/onboarding_create_profile_page.dart:63]: is this mock in purpose?

File: lib/auth/onboarding_create_profile_page.dart [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+            _messageStatusTextFieldController.text.isEmpty &&
+            _userPicture == null;
+    final _justWithUsernameAndPictureFilled =
+        _usernameTextFieldController.text.isNotEmpty &&
+            _messageStatusTextFieldController.text.isEmpty &&
+            _userPicture != null;
+    const _justUsernameFilledTitle = 'Do you want to save your changes? \n'
+        'You can always add your avatar and status message later.';
+    const _justWithUsernameAndPictureFilledTitle =
+        'Do you want to save your changes? \n'
```
Comment (stavares843) [inline lib/auth/onboarding_create_profile_page.dart:136]: can we add this to a strings file instead?

**Review by hyj1204 — APPROVED** [review]
Good job! I updated the textStyle in the TextField, right now they are good to go.

**Comment by stavares843** [comment]
neat 💥

**Pull request merged**